lib/vector-common/src/config.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::must_use_candidate)]
```

Comment:
I'm curious why this is flagged to allow, since it generally is quick to satisfy, sorta helpful.